sherpa/astro/ui/serialize.py

```diff
@@ -862,41 +968,74 @@ def _save_model_components(state, fh=None):
         #
         for par in mod.pars:
             par_attributes, par_linkstr = _print_par(par)
-            _output(par_attributes, fh)
+            _output(out, par_attributes)
             linkstr = linkstr + par_linkstr
 
         # If the model is a PSFModel, could have special
         # attributes "size" and "center" -- if so, record them.
         if typename == "psfmodel":
-            if hasattr(mod, "size"):
-                cmd = "%s.size = %s" % (modelname, repr(mod.size))
-                _output(cmd, fh)
-                _output("", fh)
-            if hasattr(mod, "center"):
-                cmd = "%s.center = %s" % (modelname, repr(mod.center))
-                _output(cmd, fh)
-                _output("", fh)
+            spacer = False
+            if hasattr(mod, "size") and mod.size is not None:
+                _output(out, f"{modelname}.size = {mod.size}")
+                spacer = True
+
+            if hasattr(mod, "center") and mod.center is not None:
+                _output(out, f"{modelname}.center = {mod.center}")
+                spacer = True
+
+            if spacer:
+                _output_nl(out)
 
     # If there were any links made between parameters, send those
     # link commands to outfile now; else, linkstr is just an empty string
-    _output(linkstr, fh)
+    _output(out, linkstr)
+    return found_xspec
+
+
+def _save_psf_components(out: OutType, state: SessionType) -> None:
+    """Save the PSF components.
+
+    This must be done after setting up the data and model components.
 
+    Parameters
+    ----------
+    out : dict
+       The output state
+    state
+
+    """
+
+    # This is done after creating all the models and datasets.
+    #
+    if len(state._psf) == 0:
+        return
+
+    # Associate any PSF models with their appropriate datasets.
+    #
+    _output_banner(out, "Associate PSF models with the datasets")
+    for idval, psfmod in state._psf.items():
+        cmd_id = _id_to_str(idval)
+        _output(out, f"set_psf({cmd_id}, {psfmod._name})")
 
-def _save_models(state, fh=None):
+
+def _save_models(out: OutType, state: SessionType) -> None:
     """Save the source, pileup, and background models.
 
     Parameters
     ----------
+    out : dict
+       The output state
     state
-    fh : None or file-like
-       If ``None``, the information is printed to standard output,
-       otherwise the information is added to the file handle.
     """
 
-    # Save all source, pileup and background models
+    ids = state.list_data_ids()
+    if len(ids) == 0:
+        return
+
+    _output_banner(out, "Set Source, Pileup and Background Models")
```

Comment:
Do we follow the Chicago manual of style?
```suggestion
    _output_banner(out, "Set Source, Pileup, and Background Models")
```

Reply:
Technically  the code always used the "eats, shoots and leaves" style, but no reason to annoy Strunk and White.